src/screens/track-screen/TrackScreenDownloadButtons.tsx

```diff
@@ -0,0 +1,156 @@
+import { useCallback } from 'react'
+
+import {
+  ButtonState,
+  ButtonType as DownloadButtonType,
+  useDownloadTrackButtons
+} from 'audius-client/src/common/hooks/useDownloadTrackButtons'
+import { Name } from 'audius-client/src/common/models/Analytics'
+import { CID, ID } from 'audius-client/src/common/models/Identifiers'
+import { User } from 'audius-client/src/common/models/User'
+import { downloadTrack } from 'audius-client/src/common/store/social/tracks/actions'
+import { View } from 'react-native'
+import { useSelector } from 'react-redux'
+
+import IconDownload from 'app/assets/images/iconDownload.svg'
+import Button, { ButtonType } from 'app/components/button'
+import LoadingSpinner from 'app/components/loading-spinner'
+import { useDispatchWeb } from 'app/hooks/useDispatchWeb'
+import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
+import { useToast } from 'app/hooks/useToast'
+import { font } from 'app/styles'
+import { makeStyles } from 'app/styles/makeStyles'
+import { make, track } from 'app/utils/analytics'
+
+export type DownloadButtonProps = {
+  state: ButtonState
+  type: DownloadButtonType
+  label: string
+  onClick?: () => void
+}
+
+export const messages = {
+  followToDownload: 'Must follow artist to download',
+  addDownloadPrefix: (label: string) => `Download ${label}`
+}
+
+const useStyles = makeStyles(({ palette }) => ({
+  buttonContainer: {
+    alignSelf: 'center',
+    marginBottom: 6
+  },
+  button: {
+    paddingVertical: 4,
+    paddingHorizontal: 8
+  },
+  icon: {
+    marginRight: 4
+  },
+  label: {
+    fontSize: 14,
+    padding: 0,
+    textTransform: 'uppercase',
+    ...font('bold'),
+    color: palette.neutralLight4
+  }
+}))
+
+const DownloadButton = ({
+  label,
+  state,
+  onClick = () => {}
+}: DownloadButtonProps) => {
+  const { toast } = useToast()
+
+  const styles = useStyles()
+  const requiresFollow = state === ButtonState.REQUIRES_FOLLOW
+  const isProcessing = state === ButtonState.PROCESSING
+  const isDisabled = state === ButtonState.PROCESSING || requiresFollow
+
+  const onPress = useCallback(() => {
```

Comment:
so not sure if this is standard, but i typically define callbacks as `handle<Action>`, so if its onPress prop, then `handlePress`. ive noticed this can help with name collisions if a prop is called `onPress` and you need to slightly modify it for the child

Reply:
Yup, I need to get in the habit of this. Will update!